intellij-plugin-structure/structure-intellij/src/main/java/com/jetbrains/plugin/structure/intellij/plugin/IdePluginImpl.kt

```diff
@@ -120,10 +120,11 @@ class IdePluginImpl : IdePlugin, StructurallyValidated {
         modulesDescriptors.addAll(old.modulesDescriptors)
         thirdPartyDependencies = old.thirdPartyDependencies.toMutableList()
         if (old is StructurallyValidated) {
-          if (structureProblems.isNotEmpty()) {
+          if (overrideStructureProblems) {
             problems.addAll(structureProblems)
           } else {
-            problems.addAll(old.problems)
+            val oldProblems = (structureProblems + old.problems).toSet()
+            problems.addAll(oldProblems)
```

Comment:
Won't there be a mess that in case of `overrideStructureProblems` is false, copied plugin will contain both original and remapped problems? 

Also, probably, it would make sense to rename `overrideStructureProblems`, because now it's a little bit strange that we override structure problems by `structureProblems`.

Reply:
Removed and simplified semantics.

The method is used in a single place with specific semantics, no need to make it more complex.